geomstats/visualization.py

```diff
@@ -490,6 +494,186 @@ def draw(self, ax, **kwargs):
+        if self.point_type == 'extrinsic':
+            if not gs.all(M32.belongs(points)):
+                raise ValueError(
+                    'Points do not belong to Matrices(3, 2).')
+            points = S32.projection(points)
+        elif self.point_type == 'pre-shape' \
+                and not gs.all(S32.belongs(points)):
+            raise ValueError('Points do not belong to the pre-shape space.')
+        points = self.convert_to_spherical_coordinates(points)
+        if not isinstance(points, list):
+            points = list(points)
+        self.points.extend(points)
+
+    def clear_points(self):
+        """Clear the points to draw."""
+        self.points = []
+
+    def draw(self, n_theta=25, n_phi=13, scale=.05):
+        """Draw the sphere regularly sampled with corresponding triangles."""
+        self.ax.set_axis_off()
+        plt.tight_layout()
+
+        coords_theta = gs.linspace(0, 2 * gs.pi, n_theta)
+        coords_phi = gs.linspace(0, gs.pi, n_phi)
+
+        coords_x = .5 * gs.outer(gs.sin(coords_phi), gs.cos(coords_theta))
+        coords_y = .5 * gs.outer(gs.sin(coords_phi), gs.sin(coords_theta))
+        coords_z = .5 * gs.outer(gs.cos(coords_phi),
+                                 gs.ones_like(coords_theta))
+
+        self.ax.plot_surface(coords_x, coords_y, coords_z,
+                             rstride=1, cstride=1, color='grey',
+                             linewidth=0, alpha=.1, zorder=-1)
+        self.ax.plot_wireframe(coords_x, coords_y, coords_z, linewidths=.6,
+                               color='grey', alpha=.6, zorder=-1)
+
+        def lim(theta):
+            return gs.pi - self.elev + (2 * self.elev - gs.pi) / gs.pi * abs(
+                self.azim - theta)
+
+        for theta in gs.linspace(0, 2 * gs.pi, n_theta // 2 + 1):
+            for phi in gs.linspace(0, gs.pi, n_phi):
+                if theta <= self.azim + gs.pi and phi <= lim(theta):
+                    self.draw_triangle(theta, phi, scale)
+                if theta > self.azim + gs.pi and phi \
+                        < lim(2 * self.azim + 2 * gs.pi - theta):
+                    self.draw_triangle(theta, phi, scale)
+
+    def draw_triangle(self, theta, phi, scale):
+        """Draw the corresponding triangle on the sphere at theta, phi."""
+        u_theta = gs.cos(theta) * self.ua + gs.sin(theta) * self.na
+        T = gs.cos(phi / 2) * self.can + gs.sin(phi / 2) * u_theta
+        T = scale * T
+        T = gs.hstack((T, .5 * gs.ones((3, 1))))
+        T = self.R_theta_phi(theta, phi) @ T.transpose()
+
+        x = gs.hstack((T[0], T[0, 0]))
+        y = gs.hstack((T[1], T[1, 0]))
+        z = gs.hstack((T[2], T[2, 0]))
+
+        self.ax.plot3D(x, y, z, 'grey', zorder=1)
+        c = ['red', 'green', 'blue']
+        for i in range(3):
+            self.ax.scatter(x[i], y[i], z[i],
+                            color=c[i], s=10, alpha=1, zorder=1)
+
+    @staticmethod
+    def R_theta_phi(theta, phi):
```

Comment:
NIT: Do not use uppercase letters in variable names and method names: Python thinks they are classes (see that the color is automatically put to orange).

Instead, you can rename:
`T` -> `triangle`
`R_theta_phi` -> `rotation_theta_phi`
etc

(using english words instead of letter notations also makes the code easier to understand).